### PR TITLE
MBS-12577 / MBS-12578: Convert cdtoc/remove to React and fix minor bugs

### PR DESCRIPTION
**Pull request by reosarevok**
### Implement MBS-12577, fix MBS-12578

See commit messages for details

On top of https://github.com/metabrainz/musicbrainz-server/pull/2626

### CODE REVIEW

File: lib/MusicBrainz/Server/Controller/CDTOC.pm [hunk 1]
```diff
@@ -70,10 +70,40 @@ sub show : Chained('load') PathPart('')
...
-    my $cdtoc_id  = $c->req->query_params->{cdtoc_id};
-    my $medium_id = $c->req->query_params->{medium_id};
+    my $cdtoc_id  = $c->req->query_params->{cdtoc_id}
+        or $self->error(
+            $c, status => HTTP_BAD_REQUEST,
+            message => l('Please provide a CD TOC ID.')
+        );
+
+    $self->error($c, status => HTTP_BAD_REQUEST,
+                 message => l('The provided CD TOC id is not valid.')
```
Comment (mwiencek) [inline lib/MusicBrainz/Server/Controller/CDTOC.pm:80]:
```suggestion
                 message => l('The provided CD TOC ID is not valid.')
```

File: lib/MusicBrainz/Server/Controller/CDTOC.pm [hunk 2]
```diff
@@ -70,10 +70,40 @@ sub show : Chained('load') PathPart('')
...
+        ) unless is_database_row_id($cdtoc_id);
+
+    my $medium_id = $c->req->query_params->{medium_id}
+        or $self->error(
+            $c, status => HTTP_BAD_REQUEST,
+            message => l('Please provide a medium ID.')
+        );
+
+    $self->error($c, status => HTTP_BAD_REQUEST,
+                 message => l('The provided medium id is not valid.')
```
Comment (mwiencek) [inline lib/MusicBrainz/Server/Controller/CDTOC.pm:90]: I guess this was an existing string which had "id" lowercase.

Comment (reosarevok) [reply lib/MusicBrainz/Server/Controller/CDTOC.pm:90]: Yes, although we could consider uppercasing them.

**Review by mwiencek — APPROVED** [review]
tested removing some discids locally, worked fine

**Pull request merged**